### PR TITLE
Rake task to send giftcards links to all customers

### PR DESCRIPTION
**Pull request by Jeremy-Xue**
Example screenshot for email sent from the seeded db (formatting is a little weird since I sent a bunch of emails beforehand to test but it shouldn't be purple/spaced in the actual email)
Screen Shot 2020-07-20 at 4 34 18 PM

Looking for suggestions on how best to word/format the email, as well as if we want to add an argument to only send emails to those who haven't already gotten the link (before the PR that changed the emails to link to the giftcard). Also, unsure how this interacts with GAM meals (do those count as gift card purchases?)

### CODE REVIEW

**Comment by justintmckibben** [comment]
cc @nanxiy do you have opinions on the language that would be used here?

File: lib/tasks/email_giftcards.rake [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+                        .distinct.pluck(:recipient_id)
+
+    unique_recipients.each do |id|
+      html = '<!DOCTYPE html>' \
+              '<html>' \
+              '<head>' \
+              "  <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />" \
+              '</head>' \
+              '<body>' \
+              '<p> Thank you for supporting local chinatown businesses! As our merchants open up, you will be able to redeem the giftcards you have purchased below.</p>'
```
Comment (justintmckibben) [inline lib/tasks/email_giftcards.rake:20]: Chinatown is a properly noun

Comment (justintmckibben) [reply lib/tasks/email_giftcards.rake:20]: I'd consider making it simpler and just removing this sentence `As our merchants open up, you will be able to redeem the giftcards you have purchased below.`

Comment (justintmckibben) [reply lib/tasks/email_giftcards.rake:20]: Thank you for supporting local Chinatown businesses! Your gift card to <merchant name> can be found here: <link to gift card>

Current balance: $XX.XX

Comment (justintmckibben) [reply lib/tasks/email_giftcards.rake:20]: But I'd just swing it by @nanxiy and see what she thinks

Comment (jacoblb64) [reply lib/tasks/email_giftcards.rake:20]: > Chinatown is a properly noun

I like how you called out his grammar with your own grammatical mistake 😂. But in all seriousness good catch

File: lib/tasks/email_giftcards.rake [hunk 2]
```diff
@@ -0,0 +1,36 @@
+# frozen_string_literal: true
+
+namespace :email_giftcards do
```
Comment (jacoblb64) [inline lib/tasks/email_giftcards.rake:3]: Optional, but if you want you can put this in the `emailer.rake` namespace where the other task was. That way the tasks can be called something like `rake emailer:vouchers_to_merchants` or `emailer:vouchers_to_customers` or maybe even just rename it `email` to make it even more succinct. I trust you to choose what's best!

File: lib/tasks/email_giftcards.rake [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+              "  <meta content='text/html; charset=UTF-8' http-equiv='Content-Type' />" \
+              '</head>' \
+              '<body>' \
+              '<p> Thank you for supporting local chinatown businesses! As our merchants open up, you will be able to redeem the giftcards you have purchased below.</p>'
+      recipient = Contact.find_by(id: id)
+      gift_cards = GiftCardDetail.where(recipient_id: id).to_a
+      html += '<style> table {border: 1px solid black}</style>'\
+            '<table>'
+      gift_cards.each do |gc|
+        gift_card_url = 'merchant.sendchinatownlove.com/voucher/' + gc.gift_card_id
```
Comment (jacoblb64) [inline lib/tasks/email_giftcards.rake:26]: @justintmckibben should this be `seller_gift_card_id`? This still confuses me about which code gets used where tbh

File: lib/tasks/email_giftcards.rake [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        amount_string = EmailManager::Sender.format_amount(amount: gc.amount)
+        item = Item.find_by(id: gc.item_id)
+        merchant_name = Seller.find_by(id: item.seller_id).name
+        html += '<tr><p>View your <b>$' + amount_string + '</b> gift card for <b>' + merchant_name + '<a href="' + gift_card_url + '"> here</a></b></p></tr>'
+      end
+      html += '</table>'
+      EmailManager::Sender.send_receipt(to: recipient.email, html: html)
```
Comment (jacoblb64) [inline lib/tasks/email_giftcards.rake:33]: Optional, but might be helpful to log some output so we can track what emails get sent to whom. I did some basic logging in the task in my last PR if you want to take a look.

**Comment by Jeremy-Xue** [comment]
Ack I just realized I never ended up merging this -- I was working on adding a bit to only send the gift card emails to people who bought them before the date we added the gift card url to emails, also I think I need to filter out GAM (?)

File: app/services/email_manager/sender.rb [hunk 5]
```diff
@@ -18,13 +18,32 @@ def self.send_receipt(to:, html:)
...
-      # like
+      # Send to Receipts Eng so that they know what their receipts in prod looks like
       RestClient.post api_url,
                       from: 'receipts@sendchinatownlove.com',
                       to: 'receipts@sendchinatownlove.com',
                       subject: 'Receipt from Send Chinatown Love',
                       html: html
     end
+
+    def self.send_receipt_with_subject(to:, html:, subject:)
```
Comment (Jeremy-Xue) [inline app/services/email_manager/sender.rb:29]: I'm wondering if this is adding too much redundancy but I think this was the easiest way to send an email with a custom subject

Comment (Jeremy-Xue) [reply app/services/email_manager/sender.rb:29]: @justintmckibben was using this to send the gift card email with a custom subject, should we use this for the passport/food crawl redemption email?

**Pull request merged**